---
title: "WLS_analysis"
author: "Kim Dill-McFarland"
date: "April 20, 2017"
output:
  html_document: 
    toc: true
    toc_float: true
  pdf_document: default
---
#Load data
##Packages
```{r}
#The vegan package provides tools for descriptive community ecology. It has most basic functions of diversity analysis, community ordination and dissimilarity analysis.
#In general, this package is used for Bray-Curtis and Jaccard analyses.
library(vegan)
#The phyloseq package seeks to address issues with multiple microbiome analysis packages by providing a set of functions that internally manage the organizing, linking, storing, and analyzing of phylogenetic sequencing data. 
#In general, this package is used for UniFrac analyses.
library(phyloseq)
#Graphing package used in phyloseq. To edit the default setting of a plot, you need to use functions in this package.
library(PMCMR)
#Graphing package used in phyloseq. To edit the default setting of a plot, you need to use functions in this package.
library(ggplot2)
#Figures
library(lattice)
library(gridExtra)
```

##Functions

* R-krusk: Function to calculate Kruskal-Wallis for OTUs and variables identified in SIMPER analysis. From [A. Steinberger](https://github.com/asteinberger9/seq_scripts)
* Multiplot: for placing 2 or more ggplot2 graphs in one figure. From [Cookbook for R](http://www.cookbook-r.com/Graphs/Multiple_graphs_on_one_page_(ggplot2)/)
* Pairwise_adonis: Pairwise PERMANOVA (adonis) function to test within categorical variables with more than 2 levels. Modified from [P. M. Arbizu](https://www.researchgate.net/post/How_can_I_do_PerMANOVA_pairwise_contrasts_in_R)

```{r}
source("R_krusk.R")
source("multiplot.R")
source("Pairwise_adonis.R")
```

##Data
Load environment from `WLS_data_manipulation.Rmd`
```{r}
load("WLS_environment_final.rdata")
```

#Gradutes-only
##Beta-diversity
Tests performed on the graduate samples to avoid possible interactions between spouses and siblings. First, check if the graduate subset is different from the other sample types to ensure the subset is a good representation of the overall data set. 
```{r}
#Test if the overall microbiota differs by groups (g, p, s, e). We do not expect significance as these groups are merely a means of ID different individuals in the dataset 
#Permutational analysis of variance (PERMANOVA)
adonis(OTU ~ Link, data=Met, method="bray", permuations=1000)
adonis(OTU ~ Link, data=Met, method="jaccard", permuations=1000)
adonis(wUF.dist ~ Link,  data=Met, permuations=1000)
adonis(uwUF.dist ~ Link,  data=Met, permuations=1000)
```

There are no differences so we move forward with the graduate subset data set. We will run a number of PERMANOVA (`adonis`) tests of variables of interest. All tests are run on Bray-Curtis, Jaccard, weighted and unweighted UniFrac distance measures. For variables with missing data, the data is subset to only `complete.cases` to avoid NA errors in `adonis`.

###Sex and age
```{r}
#Does sex/gender or age correlate to the gut microbiota?
adonis(BC.dist.g ~ gender+age, data = vars.g, permutations=1000)
adonis(J.dist.g ~ gender+age, data = vars.g, permutations=1000)
adonis(wUF.dist.g ~ gender+age, data = vars.g, permutations=1000)
adonis(uwUF.dist.g ~ gender+age, data = vars.g, permutations=1000)
```

Since sex is a significant determinant of the microbiota and can impact a number of other variables of interest, we test if our variables vary by sex.

```{r}
chisq.test(table(vars.g$gender, as.factor(vars.g$AB)))
chisq.test(table(vars.g$RU57, vars.g$gender))
chisq.test(table(vars.g$RU11, vars.g$gender))
chisq.test(table(vars.g$iq, vars.g$gender))
chisq.test(table(vars.g$yrs.edu, vars.g$gender))
chisq.test(table(vars.g$cohabYN, vars.g$gender))
chisq.test(table(vars.g$child.sum, vars.g$gender))
chisq.test(table(vars.g$groom, vars.g$gender))
chisq.test(table(vars.g$social.sum, vars.g$gender))
chisq.test(table(vars.g$dog, vars.g$gender))
chisq.test(table(vars.g$cat, vars.g$gender))
chisq.test(table(vars.g$pet.other, vars.g$gender))
chisq.test(table(vars.g$clean.house, vars.g$gender))
chisq.test(table(vars.g$bmi, vars.g$gender))
chisq.test(table(vars.g$srh11, vars.g$gender))
chisq.test(table(as.factor(vars.g$hbp11), vars.g$gender))
chisq.test(table(as.factor(vars.g$hbs11), vars.g$gender))
chisq.test(table(as.factor(vars.g$heart11), vars.g$gender))
chisq.test(table(as.factor(vars.g$arth11), vars.g$gender))
chisq.test(table(as.factor(vars.g$cancer11), vars.g$gender))
chisq.test(table(as.factor(vars.g$stroke11), vars.g$gender))
chisq.test(table(as.factor(vars.g$IBS), vars.g$gender))
chisq.test(table(vars.g$walk.ave, vars.g$gender))
chisq.test(table(as.factor(vars.g$smokeYN), vars.g$gender))
chisq.test(table(vars.g$meat, vars.g$gender))
chisq.test(table(vars.g$poul, vars.g$gender))
chisq.test(table(vars.g$pork, vars.g$gender))
chisq.test(table(vars.g$sea, vars.g$gender))
chisq.test(table(vars.g$prot.sum, vars.g$gender))
chisq.test(table(vars.g$fruit.sum, vars.g$gender))
chisq.test(table(vars.g$veg.sum, vars.g$gender))
```

And correct these P-values for multiple comparison testing.
```{r}
chi.list= c(0.8722,0.291,0.1907,0.2011,0.8032,0.0002174,0.05332,0.7207,0.9015,0.4515,0.167,0.9611,0.8052,0.3167,0.6383,5.47E-08,0.5732,0.2358,0.1249,0.362,0.4763,0.5982,0.02806,0.961,0.3057,0.8022,0.1672,0.4582,0.2996,0.3984,0.2245)

p.adjust(chi.list, method="BH")
```

For the variables that do differ by sex, we will stratify (`strata`) the model by sex. These include

* cohabYN
* walk.ave

###Antibiotics
Does antibiotic use correlate to the gut microbiota?

* AB = antibiotic use in the last 6 months, YN

```{r Antibiotics}
vars = data.frame(vars.g$AB)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(AB), data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(AB), data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(AB), data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(AB), data = vars.g[complete.cases(vars),], 
       permuations=1000)
```

###Rural vs. urban
Does rural-urban upbringing or current residence correlate to the gut microbiota?

* RU57 = OCF357 = Father's occupation in 1957
* RU11 = hf017j1e = 1990 occupation code for graduates

```{r Rural.urban}
vars = data.frame(vars.g$RU57, vars.g$RU11)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ RU57+RU11, data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ RU57+RU11, data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ RU57+RU11, data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ RU57+RU11, data = vars.g[complete.cases(vars),], 
       permuations=1000)
```

###Education
Does education or IQ correlate to the gut microbiota?

* yrs.edu = hb103red = years of education
* iq = IQ (Henmom-Nelson score)

```{r Education.IQ}
vars = data.frame(vars.g$yrs.edu, vars.g$iq)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ yrs.edu+iq, data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ yrs.edu+iq, data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ yrs.edu+iq, data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ yrs.edu+iq, data = vars.g[complete.cases(vars),], 
       permuations=1000)
```

###Human interaction
Does human interaction correlate to the gut microbiota?

* cohabYN = cohab = Cohabiting with spouse, YN (strata by sex)
* child.sum = HD01701-06 = # of children cohabiting with graduate
* groom = ha103re = Personal grooming score (0-10)
* social.sum = jz023rer + jz024rer = # of social interactions with family and friends in the last 4 wks

```{r Human.interaction}
#Full model
vars = data.frame(vars.g$cohabYN, vars.g$child.sum, vars.g$groom, vars.g$social.sum)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ cohabYN + child.sum + groom + social.sum + cohabYN:social.sum, 
       data = vars.g[complete.cases(vars),], strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ cohabYN + child.sum + groom + social.sum + cohabYN:social.sum, 
       data = vars.g[complete.cases(vars),], strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ cohabYN + child.sum + groom + social.sum + cohabYN:social.sum, 
       data = vars.g[complete.cases(vars),], strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ cohabYN + child.sum + groom + social.sum + cohabYN:social.sum, 
       data = vars.g[complete.cases(vars),], strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)

#CohabYN alone
vars = data.frame(vars.g$cohabYN)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ cohabYN, 
       data = vars.g[complete.cases(vars),], strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ cohabYN, 
       data = vars.g[complete.cases(vars),], strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ cohabYN, 
       data = vars.g[complete.cases(vars),], strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ cohabYN, 
       data = vars.g[complete.cases(vars),], strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)

#Socialness alone
##Within cohabiting persons
vars = data.frame(vars.g$social.sum)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",]), row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",])]
       ~ social.sum, 
       data = vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",], permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",]), row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",])]
       ~ social.sum, 
       data = vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",], permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",]), row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",])]
       ~ social.sum, 
       data = vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",], permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",]), row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",])]
       ~ social.sum, 
       data = vars.g[complete.cases(vars) & vars.g$cohabYN == "Y",], permuations=1000)

##Within non-cohabiting persons
vars = data.frame(vars.g$social.sum)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "N",]), row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "N",])]
       ~ social.sum, 
       data = vars.g[complete.cases(vars) & vars.g$cohabYN == "N",], permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "N",]), row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "N",])]
       ~ social.sum, 
       data = vars.g[complete.cases(vars) & vars.g$cohabYN == "N",], permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "N",]), row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "N",])]
       ~ social.sum, 
       data = vars.g[complete.cases(vars) & vars.g$cohabYN == "N",], permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "N",]), row.names(vars.g[complete.cases(vars) & vars.g$cohabYN == "N",])]
       ~ social.sum, 
       data = vars.g[complete.cases(vars) & vars.g$cohabYN == "N",], permuations=1000)
```

###Home environment
Does the home environment correlate to the gut microbiota?

* dog = dog in house
* cat = cat in house
* pet.other = Other pet (bird, reptile, fish) in house
* clean.house = ha114re = Residence cleanliness score (1-10)

```{r Home.environment}
vars = data.frame(vars.g$dog, vars.g$cat, vars.g$pet.other, vars.g$clean.house)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ dog+cat+pet.other+clean.house, data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ dog+cat+pet.other+clean.house, data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ dog+cat+pet.other+clean.house, data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ dog+cat+pet.other+clean.house, data = vars.g[complete.cases(vars),], 
       permuations=1000)
```

###Health
Does health correlate to the gut microbiota?

* bmi = body mass index based on height and weight
* srh11 = Self-reported health score (1-5)
* walk.ave = (HX472RE+HX473RE)/2 = Average walking speed (sec) (strate by sex)
* hbs11 = High-blood sugar, YN
* hbp11 = High-blood pressure, YN
* heart11 = Heart disease, YN
* arth11 = Arthritis, YN
* cancer11 = Cancer, YN
* stroke11 = Stroke, YN
* IBS = irritable bowel syndrome, YN

```{r Health}
#overall health
vars = data.frame(vars.g$bmi, vars.g$srh11, vars.g$walk.ave)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ bmi+srh11+walk.ave, data = vars.g[complete.cases(vars),], 
       strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ bmi+srh11+walk.ave, data = vars.g[complete.cases(vars),], 
       strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ bmi+srh11+walk.ave, data = vars.g[complete.cases(vars),], 
       strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ bmi+srh11+walk.ave, data = vars.g[complete.cases(vars),], 
       strata = vars.g[complete.cases(vars),]$gender,
       permuations=1000)

#Specific diseases
vars = data.frame(vars.g$hbs11, vars.g$hbp11, vars.g$heart11, vars.g$arth11, vars.g$cancer11, vars.g$stroke11, vars.g$IBS)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(hbs11) + as.factor(hbp11) + as.factor(heart11) + as.factor(arth11) + as.factor(cancer11) + as.factor(stroke11) + as.factor(IBS),
       data = vars.g[complete.cases(vars),], permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(hbs11) + as.factor(hbp11) + as.factor(heart11) + as.factor(arth11) + as.factor(cancer11) + as.factor(stroke11) + as.factor(IBS),
       data = vars.g[complete.cases(vars),], permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(hbs11) + as.factor(hbp11) + as.factor(heart11) + as.factor(arth11) + as.factor(cancer11) + as.factor(stroke11) + as.factor(IBS),
       data = vars.g[complete.cases(vars),], permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(hbs11) + as.factor(hbp11) + as.factor(heart11) + as.factor(arth11) + as.factor(cancer11) + as.factor(stroke11) + as.factor(IBS),
       data = vars.g[complete.cases(vars),], permuations=1000)
```

###Smoking
Does smoking impact the microbiota?

* smokeYN = jx013rec = currrently a smoker, YN

```{r Smoking}
vars = data.frame(vars.g$smokeYN)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(smokeYN), data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(smokeYN), data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(smokeYN), data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ as.factor(smokeYN), data = vars.g[complete.cases(vars),], 
       permuations=1000)
```

###Diet
Does diet impact the microbiota?

* prot.sum = Times consume protein / wk
* fruit.sum = # different fruits consume / week
* veg.sum = # different vegetables consume  / week
* meat = Times consume (red) meat / week
* poul = Times consume poultry / week
* pork = Times consume pork / week
* sea = Times consume seafood / week

```{r Diet}
#Protein, veg and fruits
vars = data.frame(vars.g$prot.sum, vars.g$fruit.sum, vars.g$veg.sum)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ prot.sum + fruit.sum + veg.sum, 
       data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ prot.sum + fruit.sum + veg.sum, 
       data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ prot.sum + fruit.sum + veg.sum, 
       data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ prot.sum + fruit.sum + veg.sum, 
       data = vars.g[complete.cases(vars),], 
       permuations=1000)

#Specific proteins
vars = data.frame(vars.g$meat, vars.g$poul, vars.g$pork, vars.g$sea)
adonis(BC.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ meat+poul+pork+sea+meat:poul,  
       data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(J.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ meat+poul+pork+sea+meat:poul, 
       data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(wUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ meat+poul+pork+sea+meat:poul, 
       data = vars.g[complete.cases(vars),], 
       permuations=1000)
adonis(uwUF.dist.g[row.names(vars.g[complete.cases(vars),]),row.names(vars.g[complete.cases(vars),])]
       ~ meat+poul+pork+sea+meat:poul, 
       data = vars.g[complete.cases(vars),], 
       permuations=1000)
```

###FDR-correction
Correct P-values for beta-diversity tests across all graduates for multiple comparison testing.

```{r FDR.corrections.g}
BC.list = c(0.000999,0.346653,0.002,0.726,0.17,0.277,0.297,0.125,0.551,0.517,0.623,0.003,0.026,0.061,0.019,0.902,0.792,0.678,0.658,0.056,0.275,0.679,0.007,0.397,0.007,0.915,0.649,0.913,0.19,0.351,0.011,0.674,0.045,0.413,0.092,0.067,0.107,0.004)
J.list = c(0.000999,0.323676,0.002,0.652,0.239,0.361,0.355,0.149,0.533,0.482,0.607,0.006,0.019,0.057,0.007,0.916,0.814,0.609,0.684,0.048,0.281,0.756,0.01,0.398,0.019,0.896,0.755,0.942,0.17,0.293,0.008,0.706,0.032,0.408,0.169,0.059,0.085,0.008)
wUF.list = c(0.000999,0.803197,0.005,0.82,0.239,0.525,0.369,0.14,0.553,0.628,0.567,0.023,0.045,0.262,0.046,0.83,0.574,0.873,0.691,0.272,0.471,0.808,0.021,0.592,0.053,0.959,0.477,0.779,0.302,0.563,0.006,0.19,0.154,0.047,0.09,0.028,0.368,0.006)
uwUF.list = c(0.000999,0.888112,0.001,0.863,0.032,0.142,0.738,0.212,0.108,0.541,0.273,0.001,0.046,0.066,0.005,0.931,0.806,0.218,0.505,0.053,0.376,0.272,0.003,0.272,0.009,0.502,0.87,0.815,0.342,0.418,0.067,0.664,0.033,0.411,0.149,0.091,0.19,0.004)

p.adjust(BC.list, method="BH")
p.adjust(J.list, method="BH")
p.adjust(wUF.list, method="BH")
p.adjust(uwUF.list, method="BH")
```

##Alpha-diversity
Tests performed on the graduate samples to avoid possible interactions between spouses and siblings. First, check if the graduate subset is different from the other sample types to ensure the subset is a good representation of the overall data set.

```{r}
summary(aov(shannon  ~ Link, data=Met))
summary(aov((1/simpson) ~ Link, data=Met))
summary(aov(chao  ~ Link, data=Met))
summary(aov(ace  ~ Link, data=Met))
```

There are no differences so we move forward with the graduate subset data set. We will run a number of ANOVA (`aov`) tests of variables of interest similar to those run on beta-diversity. All tests are run on the Shannon's diversity index and Chao richness estimate. For more information on variables, see the corresponding beta-diversity test.

###Sex and age

```{r Basic}
summary(aov(Met.g$shannon  ~ gender+age, data = vars.g))
summary(aov(Met.g$chao  ~ gender+age, data = vars.g))
```

###Antibiotics

```{r Antibiotics}
summary(aov(Met.g$shannon  ~ as.factor(AB), data = vars.g))
summary(aov(Met.g$chao  ~ as.factor(AB), data = vars.g))
```

###Rural vs. urban

```{r Rural.urban}
summary(aov(Met.g$shannon ~ RU57*RU15, data = vars.g))
summary(aov(Met.g$chao ~ RU57*RU15, data = vars.g))
```

###Education

```{r Education.IQ}
summary(aov(Met.g$shannon ~ yrs.edu+iq, data = vars.g))
summary(aov(Met.g$chao ~ yrs.edu+iq, data = vars.g))
```

###Human interaction

```{r Human.interaction}
summary(aov(Met.g$shannon ~ cohabYN + child.sum + groom + social.sum + cohabYN:social.sum, data = vars.g))
summary(aov(Met.g$chao ~ cohabYN + child.sum + groom + social.sum + cohabYN:social.sum, data = vars.g))

summary(aov(Met.g$shannon ~ cohabYN, data = vars.g))
summary(aov(Met.g$chao ~ cohabYN, data = vars.g))

summary(aov(Met.g[vars.g$cohabYN == "Y",]$shannon ~ social.sum, data = vars.g[vars.g$cohabYN == "Y",]))
summary(aov(Met.g[vars.g$cohabYN == "Y",]$chao ~ social.sum, data = vars.g[vars.g$cohabYN == "Y",]))

summary(aov(Met.g[vars.g$cohabYN == "N",]$shannon ~ social.sum, data = vars.g[vars.g$cohabYN == "N",]))
summary(aov(Met.g[vars.g$cohabYN == "N",]$chao ~ social.sum, data = vars.g[vars.g$cohabYN == "N",]))
```

###Home environment

```{r Home.environment}
summary(aov(Met.g$shannon ~ dog + cat + pet.other + clean.house, data = vars.g))
summary(aov(Met.g$chao ~ dog + cat + pet.other + clean.house, data = vars.g))
```

###Health 

```{r Health}
summary(aov(Met.g$shannon ~ bmi+srh11+walk.ave, data = vars.g))
summary(aov(Met.g$chao ~ bmi+srh11+walk.ave, data = vars.g))

summary(aov(Met.g$shannon ~ as.factor(hbs11)+ as.factor(hbp11)+ as.factor(heart11)+ as.factor(arth11)+ as.factor(cancer11)+ as.factor(stroke11)+ as.factor(IBS), data = vars.g))
summary(aov(Met.g$chao ~ as.factor(hbs11)+ as.factor(hbp11)+ as.factor(heart11)+ as.factor(arth11)+ as.factor(cancer11)+ as.factor(stroke11)+ as.factor(IBS), data = vars.g))
```

###Smoking

```{r Smoking}
summary(aov(Met.g$shannon ~ as.factor(smokeYN), data = vars.g))
summary(aov(Met.g$chao ~ as.factor(smokeYN), data = vars.g))
```

###Diet

```{r Diet}
summary(aov(Met.g$shannon ~ prot.sum + fruit.sum + veg.sum, data = vars.g))
summary(aov(Met.g$chao ~ prot.sum + fruit.sum + veg.sum, data = vars.g))

summary(aov(Met.g$shannon ~ meat+poul+pork+sea +meat:poul +poul:pork, data = vars.g))
summary(aov(Met.g$chao ~ meat+poul+pork+sea +meat:poul, data = vars.g))
```

###FDR-correction
```{r FDR.corrections}
shannon.list = c(0.153,0.798,0.0055,0.2075,0.1355,0.0919,0.218,0.8,0.18839,0.6775,0.4584,0.51954,0.00797,0.00516,0.222,0.0995,0.71,0.828,0.632,0.355,0.0829,0.0639,0.5078,0.0012,0.1831,0.7463,0.7133,0.4823,0.572,0.6505,0.675,0.5546,0.3442,0.0658,0.2743,0.5594,0.4588,0.8914,1.77E-06,0.0141)

chao.list = c(0.0188,0.8906,0.0191,0.122275,0.000802,0.038202,0.0143,0.3284,0.132911,0.205808,0.811443,0.685167,0.000852,0.00531,0.289,0.0529,0.246,0.42,0.535,0.832,0.0353,0.2432,0.1356,0.0422,0.3791,0.9216,0.9683,0.172,0.3964,0.9721,0.163,0.688,0.15,0.248,0.8636,0.8794,0.795,0.2378,0.0323,NA)

p.adjust(shannon.list, method="BH")
p.adjust(chao.list, method="BH")
```

#Spouses and siblings
##Beta-diversity
Are spouses or siblings more similar? Compare beta-diveristy (Bray-Curtis, Jaccard, weighted and unweighted UniFrac) of spouses, siblings, and unrelated pairs. The unrelated group is defined as pairs that were not siblings, spouses, or in-laws within the dataset. The unrelated group was subsampled to 100 pairs with sex ratios equivalent to the sibling dataset (57% same sex pairs) and P-values were averaged across 1000 iterative subsamples 

Since sample sizes and variance are unequal, compare groups by Kruskal-Wallis. Ties exist (values occur more than once in a group) so use Chi-squared method to correct within pairwise comparisons.

```{r}
KW.subsample.fxn = function(rep.number)
{library(PMCMR)
  repeat{
  BC.MF.sample = match.BC.MF[sample(1:length(match.BC.MF), 43,	replace=FALSE)]
  BC.MMFF.sample = match.BC.MMFF[sample(1:length(match.BC.MMFF), 57,	replace=FALSE)]
  J.MF.sample = match.J.MF[sample(1:length(match.J.MF), 43,	replace=FALSE)]
  J.MMFF.sample = match.J.MMFF[sample(1:length(match.J.MMFF), 57,	replace=FALSE)]
  wUF.MF.sample = match.wUF.MF[sample(1:length(match.wUF.MF), 43,	replace=FALSE)]
  wUF.MMFF.sample = match.wUF.MMFF[sample(1:length(match.wUF.MMFF), 57,	replace=FALSE)]
  uwUF.MF.sample = match.uwUF.MF[sample(1:length(match.uwUF.MF), 43,	replace=FALSE)]
  uwUF.MMFF.sample = match.uwUF.MMFF[sample(1:length(match.uwUF.MMFF), 57,	replace=FALSE)]

  dist.sample = data.frame(BC=c(match.BC.gp.se, match.BC.gs, BC.MF.sample, BC.MMFF.sample),
                   J=c(match.J.gp.se, match.J.gs, J.MF.sample, J.MMFF.sample),
                   wUF=c(match.wUF.gp.se, match.wUF.gs, wUF.MF.sample, wUF.MMFF.sample),
                   uwUF=c(match.uwUF.gp.se, match.uwUF.gs, uwUF.MF.sample, uwUF.MMFF.sample),
                   comp=c(rep(times=(length(match.BC.gp.se)), "1.gpse"), 
                               rep(times=(length(match.BC.gs)), "2.gs"),
                               rep(times=100, "3.other")))
#Bray-Curtis
BC.kw = kruskal.test(dist.sample$BC ~ dist.sample$comp)
posthoc.BC.kw.BH = p.adjust(posthoc.kruskal.nemenyi.test(x=dist.sample$BC, g=dist.sample$comp, dist="Chisq")$p.value, method="BH")
#Jaccard
J.kw = kruskal.test(dist.sample$J ~ dist.sample$comp)
posthoc.J.kw.BH = p.adjust(posthoc.kruskal.nemenyi.test(x=dist.sample$J, g=dist.sample$comp, dist="Chisq")$p.value, method="BH")
#wUF
wUF.kw = kruskal.test(dist.sample$wUF ~ dist.sample$comp)
posthoc.wUF.kw.BH = p.adjust(posthoc.kruskal.nemenyi.test(x=dist.sample$wUF, g=dist.sample$comp, dist="Chisq")$p.value, method="BH")
#uwUF
uwUF.kw = kruskal.test(dist.sample$uwUF ~ dist.sample$comp)
posthoc.uwUF.kw.BH = p.adjust(posthoc.kruskal.nemenyi.test(x=dist.sample$uwUF, g=dist.sample$comp, dist="Chisq")$p.value, method="BH")

#Print to table
kw.results = data.frame(
  BC = c(BC.MF.sample, BC.MMFF.sample, BC.kw$p.value, posthoc.BC.kw.BH),
  J = c(J.MF.sample, J.MMFF.sample, J.kw$p.value, posthoc.J.kw.BH),
  wUF = c(wUF.MF.sample, wUF.MMFF.sample, wUF.kw$p.value, posthoc.wUF.kw.BH),
  uwUF = c(uwUF.MF.sample, uwUF.MMFF.sample, uwUF.kw$p.value, posthoc.uwUF.kw.BH),
  label = c(rep("dist",100), "KW.P", "Sp.Sib", "Sp.other", "NA", "Sib.other"))

write.table(kw.results, file="KW.results.csv", append=TRUE, sep=",", col.names=FALSE)
KW.results.final = read.table("KW.results.csv", sep=",")

if(nrow(KW.results.final) >= 105*rep.number){break}
  }}

KW.subsample.fxn(1000)

#Curating results
KW.results.final = read.table("KW.results.csv", sep=",")
colnames(KW.results.final) = c("rep", "BC", "J", "wUF", "uwUF", "label")

#Average P-values
KW.P.ave = data.frame(
  KW.P = colMeans(KW.results.final[KW.results.final$label == "KW.P" & complete.cases(KW.results.final),2:5]),
  Sp.Sib = colMeans(KW.results.final[KW.results.final$label == "Sp.Sib" & complete.cases(KW.results.final),2:5]),
  Sp.other = colMeans(KW.results.final[KW.results.final$label == "Sp.other" & complete.cases(KW.results.final),2:5]),
  Sib.other = colMeans(KW.results.final[KW.results.final$label == "Sib.other"& complete.cases(KW.results.final),2:5])
)

write.table(KW.P.ave, file="KW.P.ave.csv", append=TRUE, sep=",", col.names=TRUE)
```

Use linear models to assess if beta-diversity among spouse and/or sibling pairs is explained by variables of interest. Variables found to impact the overall microbiota among graduates (PERMANOVA P < 0.1, Dataset S2), those with considerably more variation in the full dataset (age), and those specific to sibling/spouse relationships (kinship, years cohabitating, closeness) were tested.

Variables specified as `diff` are the difference between values for individuals within a pair. For example, a spousal pair of person 1 = 70 yrs old and person 2 = 75 yrs old would have an `age.diff` of 5 yrs. In this way, we are testing if pairs being discordant in one or more variables causes their microbiota to be more or less similar.

###Variables in spouses and siblings
####Education
```{r}
#Years of education
summary(lm(match.BC.gp.se.gs ~ vars.gp.se.gs$yrs.edu.diff))
summary(lm(match.J.gp.se.gs ~ vars.gp.se.gs$yrs.edu.diff))
summary(lm(match.wUF.gp.se.gs ~ vars.gp.se.gs$yrs.edu.diff))
summary(lm(match.uwUF.gp.se.gs ~ vars.gp.se.gs$yrs.edu.diff))

#IQ
summary(lm(match.BC.gp.se.gs ~ vars.gp.se.gs$iq.diff))
summary(lm(match.J.gp.se.gs ~ vars.gp.se.gs$iq.diff))
summary(lm(match.wUF.gp.se.gs ~ vars.gp.se.gs$iq.diff))
summary(lm(match.uwUF.gp.se.gs ~ vars.gp.se.gs$iq.diff))
```

####Closeness
Self-reported relationship closeness.

* 1 = not at all
* 2 = not very
* 3 = somewhat
* 4 = very

If individuals within a pair did not give the same answer, the lower of the two was used.

```{r}
summary(aov(match.BC.gp.se.gs ~ as.factor(floor(vars.gp.se.gs$closeness))))
summary(aov(match.J.gp.se.gs ~ as.factor(floor(vars.gp.se.gs$closeness))))
summary(aov(match.wUF.gp.se.gs ~ as.factor(floor(vars.gp.se.gs$closeness))))
summary(aov(match.uwUF.gp.se.gs ~ as.factor(floor(vars.gp.se.gs$closeness))))

#Pairwise between closeness levels
pairwise.t.test(match.BC.gp.se.gs, as.factor(floor(vars.gp.se.gs$closeness)), p.adj = "BH")
pairwise.t.test(match.J.gp.se.gs, as.factor(floor(vars.gp.se.gs$closeness)), p.adj = "BH")
pairwise.t.test(match.wUF.gp.se.gs, as.factor(floor(vars.gp.se.gs$closeness)), p.adj = "BH")
pairwise.t.test(match.uwUF.gp.se.gs, as.factor(floor(vars.gp.se.gs$closeness)), p.adj = "BH")

##Pairwise between closeness levels and spouses vs. siblings
vars.gp.se.gs$close.group = paste(as.factor(floor(vars.gp.se.gs$closeness)), vars.gp.se.gs$group, sep=".")

pairwise.t.test(match.BC.gp.se.gs, vars.gp.se.gs$close.group, p.adj = "BH")
pairwise.t.test(match.J.gp.se.gs, vars.gp.se.gs$close.group, p.adj = "BH")
pairwise.t.test(match.wUF.gp.se.gs, vars.gp.se.gs$close.group, p.adj = "BH")
pairwise.t.test(match.uwUF.gp.se.gs, vars.gp.se.gs$close.group, p.adj = "BH")
```

####Health
```{r}
#Heart disease
summary(aov(match.BC.gp.se.gs ~ as.factor(vars.gp.se.gs$heart.diff.cat)))
summary(aov(match.J.gp.se.gs ~ as.factor(vars.gp.se.gs$heart.diff.cat)))
summary(aov(match.wUF.gp.se.gs ~ as.factor(vars.gp.se.gs$heart.diff.cat)))
summary(aov(match.uwUF.gp.se.gs ~ as.factor(vars.gp.se.gs$heart.diff.cat)))

#High blood sugar
summary(aov(match.BC.gp.se.gs ~ as.factor(vars.gp.se.gs$hbs.diff.cat)))
summary(aov(match.J.gp.se.gs ~ as.factor(vars.gp.se.gs$hbs.diff.cat)))
summary(aov(match.wUF.gp.se.gs ~ as.factor(vars.gp.se.gs$hbs.diff.cat)))
summary(aov(match.uwUF.gp.se.gs ~ as.factor(vars.gp.se.gs$hbs.diff.cat)))
```

####Diet
```{r}
#Distance in protein consumption
summary(lm(match.BC.gp.se.gs ~ match.prot.gp.se.gs))
summary(lm(match.J.gp.se.gs ~ match.prot.gp.se.gs))
summary(lm(match.wUF.gp.se.gs ~ match.prot.gp.se.gs))
summary(lm(match.uwUF.gp.se.gs ~ match.prot.gp.se.gs))
summary(lm(match.uwUF.gp.se.gs ~ match.prot.gp.se.gs))

#Specific protein sources
summary(lm(match.BC.gp.se.gs ~  vars.gp.se.gs$meat.diff*vars.gp.se.gs$poul.diff))
summary(lm(match.J.gp.se.gs ~  vars.gp.se.gs$meat.diff*vars.gp.se.gs$poul.diff))
summary(lm(match.wUF.gp.se.gs ~  vars.gp.se.gs$meat.diff*vars.gp.se.gs$poul.diff))
summary(lm(match.uwUF.gp.se.gs ~  vars.gp.se.gs$meat.diff*vars.gp.se.gs$poul.diff))
```


###Variables only in spouses
####Rural vs. urban
Not tested for siblings as all pairs resided together as children
```{r}
summary(aov(match.BC.gp.se ~ vars.gp.se.all$RU.diff.cat))
summary(aov(match.J.gp.se ~ vars.gp.se.all$RU.diff.cat))
summary(aov(match.wUF.gp.se ~ vars.gp.se.all$RU.diff.cat))
summary(aov(match.uwUF.gp.se ~ vars.gp.se.all$RU.diff.cat))
```

####Years cohabiting
```{r}
summary(lm(match.BC.gp.se ~ vars.gp.se.all$yrs.cohab.marr))
summary(lm(match.J.gp.se ~ vars.gp.se.all$yrs.cohab.marr))
summary(lm(match.wUF.gp.se ~ vars.gp.se.all$yrs.cohab.marr))
summary(lm(match.uwUF.gp.se ~ vars.gp.se.all$yrs.cohab.marr))
```

####Closeness
Test if closeness results hold true across just spouses
```{r}
summary(lm(match.BC.gp.se ~ as.factor(floor(vars.gp.se.all$closeness))))
summary(lm(match.J.gp.se ~ as.factor(floor(vars.gp.se.all$closeness))))
summary(lm(match.wUF.gp.se ~ as.factor(floor(vars.gp.se.all$closeness))))
summary(lm(match.uwUF.gp.se ~ as.factor(floor(vars.gp.se.all$closeness))))
```

###Variables only in siblings
####Sex
Not tested for spouses because all spousal pairs were male-female
```{r}
summary(aov(match.BC.gs ~ as.factor(vars.gs.all$gender.diff.cat)))
summary(aov(match.J.gs ~ as.factor(vars.gs.all$gender.diff.cat)))
summary(aov(match.wUF.gs ~ as.factor(vars.gs.all$gender.diff.cat)))
summary(aov(match.uwUF.gs ~ as.factor(vars.gs.all$gender.diff.cat)))
```

####Genetics
Genetic relatedness defined by a kinship score.
```{r}
#Subset data to those with kinship scores available
match.BC.gs.relate = subset(match.BC.gs, names(match.BC.gs) %in% row.names(relatedness.gs.match))
match.J.gs.relate = subset(match.J.gs, names(match.J.gs) %in% row.names(relatedness.gs.match))
match.wUF.gs.relate = subset(match.wUF.gs, names(match.wUF.gs) %in% row.names(relatedness.gs.match))
match.uwUF.gs.relate = subset(match.uwUF.gs, names(match.uwUF.gs) %in% row.names(relatedness.gs.match))

summary(aov(match.BC.gs.relate ~ relatedness.gs.match$kinship))
summary(aov(match.J.gs.relate ~ relatedness.gs.match$kinship))
summary(aov(match.wUF.gs.relate ~ relatedness.gs.match$kinship))
summary(aov(match.uwUF.gs.relate ~ relatedness.gs.match$kinship))
```

####Rural vs. urban
Not tested for spouses as all resided together at time of sampling.
```{r}
summary(aov(match.BC.gs ~ as.factor(vars.gs.all$RU11.diff)))
summary(aov(match.J.gs ~ as.factor(vars.gs.all$RU11.diff)))
summary(aov(match.wUF.gs ~ as.factor(vars.gs.all$RU11.diff)))
summary(aov(match.uwUF.gs ~ as.factor(vars.gs.all$RU11.diff)))
```

####CohabYN
Cohabiting with a spouse or not
```{r}
summary(aov(match.BC.gs ~ as.factor(vars.gs.all$cohab.diff.cat)))
summary(aov(match.J.gs ~ as.factor(vars.gs.all$cohab.diff.cat)))
summary(aov(match.wUF.gs ~ as.factor(vars.gs.all$cohab.diff.cat)))
summary(aov(match.uwUF.gs ~ as.factor(vars.gs.all$cohab.diff.cat)))
```

####Socialness
```{r}
summary(lm(match.BC.gs ~ vars.gs.all$social.sum.diff))
summary(lm(match.J.gs ~ vars.gs.all$social.sum.diff))
summary(lm(match.wUF.gs ~ vars.gs.all$social.sum.diff))
summary(lm(match.uwUF.gs ~ vars.gs.all$social.sum.diff))
```

####Closeness
Test if closeness results hold true across just siblings
```{r}
summary(aov(match.BC.gs ~ as.factor(floor(vars.gs.all$closeness.ave))))
summary(aov(match.J.gs ~ as.factor(floor(vars.gs.all$closeness.ave))))
summary(aov(match.wUF.gs ~ as.factor(floor(vars.gs.all$closeness.ave))))
summary(aov(match.uwUF.gs ~ as.factor(floor(vars.gs.all$closeness.ave))))
```

###FDR-correction
```{r}
BC.sib.spouse.list = c(0.554,0.257,0.869,0.514,0.271,0.0626,0.118,0.0319,0.544,0.272,0.194,0.0148,0.272,0.5318,0.7282,0.0318)
J.sib.spouse.list = c(0.491,0.219,0.871,0.414,0.254,0.0533,0.155,0.0305,0.519,0.219,0.159,0.0179,0.301,0.5381,0.7658,0.0428)
wUF.sib.spouse.list = c(0.338,0.657,0.901,0.829,0.832,0.0717,0.176,0.0562,0.962,0.754,0.147,0.052,0.672,0.6297,0.3804,0.0358)
uwUF.sib.spouse.list = c(0.361,0.0512,0.609,0.911,0.427,0.177,0.61,0.0266,0.731,0.00405,0.424,0.106,0.0119,0.813,0.302,0.161)

p.adjust(BC.sib.spouse.list, method="BH")
p.adjust(J.sib.spouse.list, method="BH")
p.adjust(wUF.sib.spouse.list, method="BH")
p.adjust(uwUF.sib.spouse.list, method="BH")
```

##Interactions
Do variables that explain beta-diversity differ between spouse and sibling groups?
```{r}
#AB
chisq.test(table(vars.gp.se.gs$group, as.factor(vars.gp.se.gs$AB.diff.cat)))
#Closeness
chisq.test(table(vars.gp.se.gs$group, as.factor(floor(vars.gp.se.gs$closeness))))
#heart.diff
chisq.test(table(vars.gp.se.gs$group, as.factor(vars.gp.se.gs$heart.diff.cat)))
#hbs.diff
chisq.test(table(vars.gp.se.gs$group, as.factor(vars.gp.se.gs$hbs.diff.cat)))
#dist.prot
summary(aov(match.prot.gp.se.gs ~ vars.gp.se.gs$group))
#meat.diff:poul.diff
summary(aov(vars.gp.se.gs$meat.diff ~ vars.gp.se.gs$group))
summary(aov(vars.gp.se.gs$poul.diff ~ vars.gp.se.gs$group))
```

FDR-correction
```{r}
p.adjust(c(0.1921,7.14E-13,0.7974,1,8.99E-10,1.99E-07,1.61E-05), method="BH")
```

It appears that spouses tend to have closer relationships and more similar diets than siblings.
```{r}
#Plots
par(mfrow = c(1, 2))
plot(table(vars.gp.se.gs$group, as.factor(floor(vars.gp.se.gs$closeness))), ylab="closeness")
boxplot(match.prot.gp.se.gs ~ vars.gp.se.gs$group, ylab="dist.prot")
```

#OTUs and health
OTUs commonly shared between siblings or spouses (> 50% of pairs) and rare in the overall dataset (< 70% of samples, < 49% of unrelated pairs) were correlated to health outcomes in the whole dataset by Kruskal-Wallis with the Benjamini-Hochberg correction for multiple comparisons.

We will co-op the `kruskal.pretty` function to run these tests by creating a fake SIMPER output. We will also need to load our unaltered `.cons.taxonomy` table from mothur as the function cannot take the altered version `Tax` we have used those far.

```{r}
#Fake SIMPER output to get OTU list for health analyses
fake.simper = data.frame(
  X = c(1:22),
  Comparison = rep("1_2", 22),
  SIMPER = rep(0.001, 22),
  OTU = c("Otu00027","Otu00035","Otu00043","Otu00007","Otu00019","Otu00025","Otu00047","Otu00004","Otu00048","Otu00049","Otu00105","Otu00145","Otu00130","Otu00028","Otu00135","Otu00074","Otu00295","Otu00160","Otu00173","Otu00157","Otu00330","Otu00116")
)

#Unaltered mothur taxonomy table
Tax.orig = read.table("WLS.final.an.unique_list.0.02.cons.taxonomy", header=TRUE, row.names=1, sep="\t")
```

We then run Kruskal-Wallis tests for our health variables with NAs removed.
```{r}
kruskal.pretty(OTU[!is.na(Pheno.sm$hbp11),], Pheno.sm[!is.na(Pheno.sm$hbp11),], fake.simper, c("hbp11"), 'hbp', Tax.orig)
kruskal.pretty(OTU[!is.na(Pheno.sm$hbs11),], Pheno.sm[!is.na(Pheno.sm$hbs11),], fake.simper, c("hbs11"), 'hbs', Tax.orig)
kruskal.pretty(OTU[!is.na(Pheno.sm$heart11),], Pheno.sm[!is.na(Pheno.sm$heart11),], fake.simper, c("heart11"), 'heart', Tax.orig)
kruskal.pretty(OTU[!is.na(Pheno.sm$arth11),], Pheno.sm[!is.na(Pheno.sm$arth11),], fake.simper, c("arth11"), 'arth', Tax.orig)
kruskal.pretty(OTU[!is.na(Pheno.sm$cancer11),], Pheno.sm[!is.na(Pheno.sm$cancer11),], fake.simper, c("cancer11"), 'cancer', Tax.orig)
kruskal.pretty(OTU[!is.na(Pheno.sm$stroke11),], Pheno.sm[!is.na(Pheno.sm$stroke11),], fake.simper, c("stroke11"), 'stroke', Tax.orig)
```

The IBS variable was not part of the Pheno.sm survey so we curate it from the larger Pheno data set.

* jx148rer = graduates
* px148rer = siblings

```{r}
vars.IBS = data.frame(
  IBS = c(Pheno$jx148rer, Pheno$px148rer),
  ID = c(paste(Pheno$idpriv, "g", sep=""), paste(Pheno$idpriv, "s", sep="")),
  row.names = c(paste(Pheno$idpriv, "g", sep=""), paste(Pheno$idpriv, "s", sep=""))
)

#Subset to data with microbiota samples and IBS data
vars.IBS.sub = as.data.frame(subset(vars.IBS, row.names(vars.IBS) %in% row.names(OTU)))
vars.IBS.all = as.data.frame(vars.IBS.sub[!is.na(vars.IBS.sub$IBS),])
OTU.IBS = subset(OTU, row.names(OTU) %in% row.names(vars.IBS.all))
```

Run Kruskal-Wallis test of new IBS varibale.
```{r}
kruskal.pretty(OTU.IBS, vars.IBS.all, fake.simper, c("IBS"), 'IBS', Tax.orig)
```

#Figures
##Figure 1
```{r}
set.seed(51234)
#Calculate NMDS
uwUF.g.ordu = ordinate(physeq.tree.g, method="NMDS", distance="unifrac", weighted=FALSE)

#Create data set of just significant variables
vars.envfit = data.frame(
  AB = ifelse(vars.g$AB == "Y", 2, 1), #Reverse code so that arrow points toward Y responses
  cohabYN = ifelse(vars.g$cohabYN == "N", 1, 2),#Reverse code so that arrow points toward Y responses
  socialness = vars.g$social.sum,
  hbs = ifelse(vars.g$hbs11 == "Y", 2, 1),#Reverse code so that arrow points toward Y responses
  heart = ifelse(vars.g$heart11 == 1, 2, 1),#Reverse code so that arrow points toward Y responses
  meat = vars.g$meat,
  poultry = vars.g$poul,
  prot.sum = vars.g$prot.sum,
  social.cohabY = ifelse(vars.g$cohabYN == "Y", vars.g$social.sum, NA),
  social.cohabN = ifelse(vars.g$cohabYN == "N", vars.g$social.sum, NA),
  row.names = row.names(vars.g))


#Fit variables to NMDS
ef.uwUF.AB = envfit(uwUF.g.ordu ~ AB, data=vars.envfit, na.rm=TRUE)
ef.uwUF.socY = envfit(uwUF.g.ordu ~ social.cohabY, data=vars.envfit, na.rm=TRUE)
ef.uwUF.socN = envfit(uwUF.g.ordu ~ social.cohabN, data=vars.envfit, na.rm=TRUE)
ef.uwUF.hbs = envfit(uwUF.g.ordu ~ hbs, data=vars.envfit, na.rm=TRUE)
ef.uwUF.heart = envfit(uwUF.g.ordu ~ heart, data=vars.envfit, na.rm=TRUE)
ef.uwUF.meat = envfit(uwUF.g.ordu ~ meat, data=vars.envfit, na.rm=TRUE)
ef.uwUF.poul = envfit(uwUF.g.ordu ~ poultry, data=vars.envfit, na.rm=TRUE)

#Figure 1
postscript("FIGURE1.eps", width = 4.2, height = 4.5, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
plot(uwUF.g.ordu, type="n", main="", xlab="", ylab="")
points(uwUF.g.ordu, pch=21, cex=0.75, display="sites", col="black", bg=c("red","blue")[as.factor(Pheno.sm.g$gender)])
plot(ef.uwUF.AB, col="red")
plot(ef.uwUF.socY, col="black")
plot(ef.uwUF.socN, col="red")
plot(ef.uwUF.hbs, col="red")
plot(ef.uwUF.heart, col="red")
plot(ef.uwUF.meat, col="black")
plot(ef.uwUF.poul, col="black")
dev.off()
```

##Figure 2
```{r}
#Figure 2
postscript("FIGURE2.eps", width = 5, height = 3, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
layout(matrix(c(1,2), 1, 2, byrow = TRUE), widths=c(1,1), heights=c(1,1))

#Shannon
boxplot(Met.g$shannon ~ vars.g$cohabYN, ylab="Shannon's diversity", main="", xlab="Cohabitating", col=c("red", "blue"))
stripchart(Met.g$shannon ~ vars.g$cohabYN, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.25)

#Chao
boxplot(Met.g$chao ~ vars.g$cohabYN, ylab="Chao's richness", main="", xlab="Cohabitating", col=c("red", "blue"))
stripchart(Met.g$chao ~ vars.g$cohabYN, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.25)
dev.off()
```

##Figure 3
```{r}
#Load results from KW.subsample.fxn
KW.results.final = read.table("KW.results.csv", sep=",")

#Re-organize into data frame
gp.se.gs.boxplot = data.frame(
  BC = c(match.BC.gp.se.gs, KW.results.final[KW.results.final$V6 == "dist" & complete.cases(KW.results.final),]$V2),
  J = c(match.J.gp.se.gs, KW.results.final[KW.results.final$V6 == "dist" & complete.cases(KW.results.final),]$V3),
  wUF = c(match.wUF.gp.se.gs, KW.results.final[KW.results.final$V6 == "dist" & complete.cases(KW.results.final),]$V4),
  uwUF = c(match.uwUF.gp.se.gs, KW.results.final[KW.results.final$V6 == "dist" & complete.cases(KW.results.final),]$V5),
  group = c(rep("1.spouse", length(match.BC.gp.se)), rep("2.sibling", length(match.BC.gs)), rep("3.unrelated", 100000))
)

#Add variable for closeness:pair type
vars.gp.se.gs$closeness.group = ifelse(floor(vars.gp.se.gs$closeness) == 1 & vars.gp.se.gs$group == "spouse", "1.spouse",
                                ifelse(floor(vars.gp.se.gs$closeness) == 2 & vars.gp.se.gs$group == "spouse", "2.spouse",
                                ifelse(floor(vars.gp.se.gs$closeness) == 3 & vars.gp.se.gs$group == "spouse", "3.spouse", 
                                ifelse(floor(vars.gp.se.gs$closeness) == 4 & vars.gp.se.gs$group == "spouse", "4.spouse",       
                                ifelse(floor(vars.gp.se.gs$closeness) == 1 & vars.gp.se.gs$group == "sib", "1.sib",       
                                ifelse(floor(vars.gp.se.gs$closeness) == 2 & vars.gp.se.gs$group == "sib", "2.sib",       
                                ifelse(floor(vars.gp.se.gs$closeness) == 3 & vars.gp.se.gs$group == "sib", "3.sib",       
                                ifelse(floor(vars.gp.se.gs$closeness) == 4 & vars.gp.se.gs$group == "sib", "4.sib", floor(vars.gp.se.gs$closeness)))))))))    

#Figure 3
postscript("FIGURE3.eps", width = 8, height = 5, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
layout(matrix(c(1,2,3), 1, 3, byrow = TRUE), widths=c(2,2.5,3), heights=c(1,1,1))

boxplot(gp.se.gs.boxplot$uwUF ~ gp.se.gs.boxplot$group, main = "", ylab="Unweighted UniFrac distance", col=c("red","blue","white"), ylim=c(0.3,0.95))

boxplot(match.uwUF.gp.se.gs~floor(vars.gp.se.gs$closeness), xlab="", ylab="", main="",  outpch=NA, ylim=c(0.3,0.95), col="purple")
stripchart(match.uwUF.gp.se.gs~floor(vars.gp.se.gs$closeness), col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.5)

boxplot(match.uwUF.gp.se.gs~vars.gp.se.gs$closeness.group, col=c("blue", "blue", "blue", "red", "blue", "red"), xlab="", ylab="", main="",  outpch=NA, ylim=c(0.3,0.95))
stripchart(match.uwUF.gp.se.gs~vars.gp.se.gs$closeness.group, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.5)

dev.off()
```

##Figure 4
```{r}
#Calculate relative abundances
abund = OTU/rowSums(OTU)

#Re-code variable to YN letters
Pheno.sm$hbs = ifelse(Pheno.sm$hbs11 == 1, "Y", ifelse(Pheno.sm$hbs11 == 2, "N", Pheno.sm$hbs11))

#Combine OTU abundance with health data
hbs.OTUs1 = data.frame(
  abundance = c(abund$Otu00160*100,abund$Otu00295*100),
  OTU = c(rep("OTU160", 408),rep("OTU295", 408)),
  hbs = rep(Pheno.sm$hbs, 2))
hbs.OTUs2 = data.frame(
  abundance = c(abund$Otu00004*100,abund$Otu00019*100,abund$Otu00074*100),
  OTU = c(rep("OTU004", 408),rep("OTU019", 408),rep("OTU074", 408)),
  hbs = rep(Pheno.sm$hbs, 3))

#Calculate means and standard deviations of OTU abundances within health groups
hbs.OTUs.data1 = aggregate(hbs.OTUs1$abundance, by=list(hbs.OTUs1$OTU, hbs.OTUs1$hbs), FUN = function(x) c(mean = mean(x), sd = sd(x), n=length(x)))
hbs.OTUs.data1 = do.call(data.frame, hbs.OTUs.data1)
hbs.OTUs.data1$se = hbs.OTUs.data1$x.sd / sqrt(hbs.OTUs.data1$x.n)
colnames(hbs.OTUs.data1) = c("OTU", "hbs", "mean", "sd", "n", "se")

hbs.OTUs.data2 = aggregate(hbs.OTUs2$abundance, by=list(hbs.OTUs2$OTU, hbs.OTUs2$hbs), FUN = function(x) c(mean = mean(x), sd = sd(x), n=length(x)))
hbs.OTUs.data2 = do.call(data.frame, hbs.OTUs.data2)
hbs.OTUs.data2$se = hbs.OTUs.data2$x.sd / sqrt(hbs.OTUs.data2$x.n)
colnames(hbs.OTUs.data2) = c("OTU", "hbs", "mean", "sd", "n", "se")

#Figure 4
limits1 <- aes(ymax = hbs.OTUs.data1$mean + hbs.OTUs.data1$se, ymin = hbs.OTUs.data1$mean - hbs.OTUs.data1$se)
p1 = ggplot(data = hbs.OTUs.data1, aes(x = factor(OTU), y = mean, fill = factor(hbs)))
limits2 <- aes(ymax = hbs.OTUs.data2$mean + hbs.OTUs.data2$se, ymin = hbs.OTUs.data2$mean - hbs.OTUs.data2$se)
p2 = ggplot(data = hbs.OTUs.data2, aes(x = factor(OTU), y = mean, fill = factor(hbs)))
plot1 = p1 + geom_bar(stat = "identity", position = position_dodge(0.9)) +
  geom_errorbar(limits1, position = position_dodge(0.9),  width = 0.25) +
  labs(x = "", y = "Relative abundance (%)") + scale_fill_manual(name = "HBS", values=c("red", "blue")) + 
  theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.text = element_text(colour="black"), axis.ticks = element_line(colour="black"))
plot2 = p2 + geom_bar(stat = "identity", position = position_dodge(0.9)) +
  geom_errorbar(limits2, position = position_dodge(0.9),  width = 0.25) +
  labs(x = "", y = "") + scale_fill_manual(name = "HBS", values=c("red", "blue")) + 
  theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.text = element_text(colour="black"), axis.ticks = element_line(colour="black"))

postscript("FIGURE4.eps", width = 5, height = 2.5, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
multiplot(plot1, plot2, cols=2)
dev.off()
```

##Figure S1
```{r}
set.seed(51234)
#Calculate NMDS
BC.g.MDS=metaMDS(OTU.g, distance="bray", k=2, trymax=500)
J.g.MDS=metaMDS(OTU.g, distance="jaccard", k=2, trymax=500)
wUF.g.ordu = ordinate(physeq.tree.g, method="NMDS", distance="unifrac", weighted=TRUE)

#Fit environmental variables
ef.BC.AB = envfit(BC.g.MDS ~ AB, data=vars.envfit, na.rm=TRUE)
ef.BC.socY = envfit(BC.g.MDS ~ social.cohabY, data=vars.envfit, na.rm=TRUE)
ef.BC.socN = envfit(BC.g.MDS ~ social.cohabN, data=vars.envfit, na.rm=TRUE)
ef.BC.hbs = envfit(BC.g.MDS ~ hbs, data=vars.envfit, na.rm=TRUE)
ef.BC.heart = envfit(BC.g.MDS ~ heart, data=vars.envfit, na.rm=TRUE)
ef.BC.prot = envfit(BC.g.MDS ~ prot.sum, data=vars.envfit, na.rm=TRUE)
ef.BC.meat = envfit(BC.g.MDS ~ meat, data=vars.envfit, na.rm=TRUE)
ef.BC.poul = envfit(BC.g.MDS ~ poultry, data=vars.envfit, na.rm=TRUE)

ef.J.AB = envfit(J.g.MDS ~ AB, data=vars.envfit, na.rm=TRUE)
ef.J.cohab = envfit(J.g.MDS ~ cohabYN, data=vars.envfit, na.rm=TRUE)
ef.J.socY = envfit(J.g.MDS ~ social.cohabY, data=vars.envfit, na.rm=TRUE)
ef.J.socN = envfit(J.g.MDS ~ social.cohabN, data=vars.envfit, na.rm=TRUE)
ef.J.hbs = envfit(J.g.MDS ~ hbs, data=vars.envfit, na.rm=TRUE)
ef.J.heart = envfit(J.g.MDS ~ heart, data=vars.envfit, na.rm=TRUE)
ef.J.prot = envfit(J.g.MDS ~ prot.sum, data=vars.envfit, na.rm=TRUE)
ef.J.meat = envfit(J.g.MDS ~ meat, data=vars.envfit, na.rm=TRUE)
ef.J.poul = envfit(J.g.MDS ~ poultry, data=vars.envfit, na.rm=TRUE)
                 
ef.wUF.AB = envfit(wUF.g.ordu ~ AB, data=vars.envfit, na.rm=TRUE)
ef.wUF.prot = envfit(wUF.g.ordu ~ prot.sum, data=vars.envfit, na.rm=TRUE)
ef.wUF.meat = envfit(wUF.g.ordu ~ meat, data=vars.envfit, na.rm=TRUE)
ef.wUF.poul = envfit(wUF.g.ordu ~ poultry, data=vars.envfit, na.rm=TRUE)

#Figure S1
postscript("FIGURES1.eps", width = 9, height = 3, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
layout(matrix(c(1,2,3), 1, 3, byrow = TRUE), widths=c(1,1,1), heights=c(1,1,1))
plot(BC.g.MDS, type="n", main="Bray-Curtis", xlab="", ylab="")
  points(BC.g.MDS, pch=21, cex=0.75, display="sites", col="black", bg=c("red","blue")[as.factor(Pheno.sm.g$gender)])
  plot(ef.BC.AB, col="red")
  plot(ef.BC.hbs, col="red")
  plot(ef.BC.heart, col="red")
  plot(ef.BC.prot, col="black")
  plot(ef.BC.socY, col="black")
  plot(ef.BC.socN, col="black")
  plot(ef.BC.meat, col="black")
  plot(ef.BC.poul, col="black")
plot(J.g.MDS, type="n", main="Jaccard", xlab="", ylab="")
  points(J.g.MDS, pch=21, cex=0.75, display="sites", col="black", bg=c("red","blue")[as.factor(Pheno.sm.g$gender)])
  plot(ef.J.AB, col="red")
  plot(ef.J.cohab, col="black")
  plot(ef.J.socY, col="black")
  plot(ef.J.socN, col="black")
  plot(ef.J.hbs, col="black")
  plot(ef.J.heart, col="black")
  plot(ef.J.prot, col="black")
  plot(ef.J.meat, col="black")
  plot(ef.J.poul, col="black")
plot(wUF.g.ordu, type="n", main="Weighted UniFrac", xlab="", ylab="", xlim=c(-0.03,0.03), ylim=c(-0.03,0.03))
  points(wUF.g.ordu, pch=21, cex=0.75, display="sites", col="black", bg=c("red","blue")[as.factor(Pheno.sm.g$gender)])
  plot(ef.wUF.AB, col="black")
  plot(ef.wUF.prot, col="black")
  plot(ef.wUF.meat, col="black")
  plot(ef.wUF.poul, col="black")
dev.off()
```

##Figure S2
```{r}
#Re-code variables to YN letters
vars.g$AB = ifelse(vars.g$AB == 1, "Y", ifelse(vars.g$AB == 2, "N", vars.g$AB))
vars.g$hbs11 = ifelse(vars.g$hbs11 == 1, "Y", ifelse(vars.g$hbs11 == 2, "N", vars.g$hbs11))

#Figure S2ABCD
postscript("FIGURES2a.eps", width = 6, height = 6, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
layout(matrix(c(1,2,3,4), 2, 2, byrow = TRUE), widths=c(1,1), heights=c(1,1))
#Shannon
boxplot(Met.g$shannon ~ vars.g$AB, ylab="Shannon's diversity", main="", xlab="Antibiotic use last 6mo")
  stripchart(Met.g$shannon ~ vars.g$AB, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.25)
boxplot(Met.g$shannon ~ vars.g$hbs11, main="", xlab="High blood sugar")
  stripchart(Met.g$shannon ~ vars.g$hbs11, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.25)
#Chao
boxplot(Met.g$chao ~ vars.g$RU57, ylab="Chao's richness", main="", xlab="1957")
  stripchart(Met.g$chao ~ vars.g$RU57, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.25)
boxplot(Met.g$chao ~ vars.g$RU11, main="", xlab="2011")
  stripchart(Met.g$chao ~ vars.g$RU11, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.25)
dev.off()

#Figure S2E
postscript("FIGURES2b.eps", width = 3, height = 3, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
layout(matrix(c(1), 1, 1, byrow = TRUE), widths=c(1), heights=c(1))
levelplot(Met.g$shannon ~ vars.g$poul*vars.g$meat, col.regions=function(x)rev(gray.colors(x)), xlab="Poultry (times per week)", ylab="Meat (times per week)", main="")
dev.off()

#Figure S2F
postscript("FIGURES2c.eps", width = 3, height = 3, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
layout(matrix(c(1), 1, 1, byrow = TRUE), widths=c(1), heights=c(1))
levelplot(Met.g$shannon ~ vars.g$poul*vars.g$pork, col.regions=function(x)rev(gray.colors(x)), xlab="Poultry (times per week)", ylab="Pork (times per week)", main="")
dev.off()
```

##Figure S3
```{r}
#Calculate linear models
shan.social.cohabY = glm(Met.g[vars.g$cohabYN == "Y",]$shannon ~ vars.g[vars.g$cohabYN == "Y",]$social.sum)
shan.social.cohabN = glm(Met.g[vars.g$cohabYN == "N",]$shannon ~ vars.g[vars.g$cohabYN == "N",]$social.sum)
chao.social.cohabY = glm(Met.g[vars.g$cohabYN == "Y",]$chao ~ vars.g[vars.g$cohabYN == "Y",]$social.sum)
chao.social.cohabN = glm(Met.g[vars.g$cohabYN == "N",]$chao ~ vars.g[vars.g$cohabYN == "N",]$social.sum)

#Figure S3
postscript("FIGURES3.eps", width = 6, height = 6, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
layout(matrix(c(1,2,3,4), 2, 2, byrow = TRUE), widths=c(1,1), heights=c(1,1))

plot(Met.g$shannon ~ vars.g$social.sum, pch=21, col="black", bg=c("red","blue")[vars.g$cohabYN], ylab="", main="", xlab="Social interactions / 4 weeks", cex=0.75)
        abline(shan.social.cohabY, col="blue")
        abline(shan.social.cohabN, col="red")
#Chao
plot(Met.g$chao ~ vars.g$social.sum, pch=21, col="black", bg=c("red","blue")[vars.g$cohabYN], ylab="", main="", xlab="Social interactions / 4 weeks", cex=0.75)
        abline(chao.social.cohabY, col="blue")
        abline(chao.social.cohabN, col="red")
dev.off()
```

##Figure S4
```{r}
#Figure S4
postscript("FIGURES4.eps", width = 9, height = 9, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
layout(matrix(c(1,2,3,4,5,6,7,8,9), 3, 3, byrow = TRUE), widths=c(1,1,1), heights=c(1,1,1))
#BC
boxplot(gp.se.gs.boxplot$BC ~ gp.se.gs.boxplot$group, main = "", ylab="Bray-Curtis distance", col=c("red","blue","white"), ylim=c(0.05,1))
boxplot(match.BC.gp.se.gs~floor(vars.gp.se.gs$closeness), xlab="", ylab="", main="",  outpch=NA, ylim=c(0.05,1), col="purple")
  stripchart(match.BC.gp.se.gs~floor(vars.gp.se.gs$closeness), col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.5)
boxplot(match.BC.gp.se.gs~vars.gp.se.gs$closeness.group, col=c("blue", "blue", "blue", "red", "blue", "red"), xlab="", ylab="", main="",  outpch=NA, ylim=c(0.05,1))
  stripchart(match.BC.gp.se.gs~vars.gp.se.gs$closeness.group, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.5)
#J
boxplot(gp.se.gs.boxplot$J ~ gp.se.gs.boxplot$group, main = "", ylab="Jaccard distance", col=c("red","blue","white"), ylim=c(0.1,1))
boxplot(match.J.gp.se.gs~floor(vars.gp.se.gs$closeness), xlab="", ylab="", main="",  outpch=NA, ylim=c(0.1,1), col="purple")
  stripchart(match.J.gp.se.gs~floor(vars.gp.se.gs$closeness), col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.5)
boxplot(match.J.gp.se.gs~vars.gp.se.gs$closeness.group, col=c("blue", "blue", "blue", "red", "blue", "red"), xlab="", ylab="", main="",  outpch=NA, ylim=c(0.1,1))
  stripchart(match.J.gp.se.gs~vars.gp.se.gs$closeness.group, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.5)
#wUF
boxplot(gp.se.gs.boxplot$wUF ~ gp.se.gs.boxplot$group, main = "", ylab="Weighted UniFrac distance", col=c("red","blue","white"), ylim=c(0.05,0.7))
boxplot(match.wUF.gp.se.gs~floor(vars.gp.se.gs$closeness), xlab="", ylab="", main="",  outpch=NA, ylim=c(0.05,0.7), col="purple")
  stripchart(match.wUF.gp.se.gs~floor(vars.gp.se.gs$closeness), col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.5)
boxplot(match.wUF.gp.se.gs~vars.gp.se.gs$closeness.group, col=c("blue", "blue", "blue", "red", "blue", "red"), xlab="", ylab="", main="",  outpch=NA, ylim=c(0.05,0.7))
  stripchart(match.wUF.gp.se.gs~vars.gp.se.gs$closeness.group, col="black", bg="black", pch=21, vertical=TRUE, method="jitter", add=TRUE, cex=0.5)
dev.off()
```

##Figure S5
```{r}
#Figure S5
postscript("FIGURES5.eps", width = 3, height = 3, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
plot(match.uwUF.gp.se.gs ~ match.prot.gp.se.gs, bg=c(rep("red",94), rep("blue", 83)), col="black", pch=21, main = "", xlab="Euclidean distances of protein consumption", ylab = "Unweighted UniFrac distance", cex=0.5)
  uwUF.spouse = lm(match.uwUF.gp.se.gs[vars.gp.se.gs$group == "spouse"] ~ match.prot.gp.se.gs[vars.gp.se.gs$group == "spouse"])
    uwUF.sib = lm(match.uwUF.gp.se.gs[vars.gp.se.gs$group == "sib"] ~ match.prot.gp.se.gs[vars.gp.se.gs$group == "sib"])
    uwUF.all = lm(match.uwUF.gp.se.gs ~ match.prot.gp.se.gs)    
    abline(uwUF.spouse, col="red")
    abline(uwUF.sib, col="blue")
    abline(uwUF.all, col="black")
dev.off()
```

##Figure S6
```{r}
#Figure S6A
postscript("FIGURES6a.eps", width = 3, height = 3, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
boxplot(match.prot.gp.se.gs~vars.gp.se.gs$group, ylab="Euclidean distances of protein consumption", main="")
dev.off()

#Figure S6B
postscript("FIGURES6b.eps", width = 3, height = 3, horizontal = FALSE, onefile = FALSE, paper = "special", colormodel = "rgb", family = "ArialMT")
plot(table(vars.gp.se.gs$group,floor(vars.gp.se.gs$closeness)), ylab="Closeness", main="")
dev.off()
```